benchmark.py

```diff
@@ -153,14 +152,14 @@ def cd(newdir):
 
 
 @contextmanager
-def repo(repository, branch=None):
+def repo(repository, name, branch=None):
     """
     cd into local copy of repository.  if the repository has not been
     cloned yet, then clone it to working directory first.
     """
     prev_dir = os.getcwd()
 
-    repo_dir = os.path.expanduser(conf["repo_dir"])
+    repo_dir= os.path.expanduser(os.path.join(conf["working_dir"], (name +"_repos")))
```

Comment:
this decorator is actually used for all the repos (project, trigger and dependency) so you don't want a separate repos_dir each time the decorator is used...  I think you want to determine the repos_dir name from the project name at the benchmark level and pass it down through the decorator